test/kitchen/drivers/azure-driver.yml

```diff
@@ -0,0 +1,130 @@
+<%
+  ENV['AZURE_LOCATION'] ||= "North Central US"
+  location = ENV['AZURE_LOCATION']
+%>
+
+<%
+  vm_tags = {
+    'dd_agent_testing': 'dd_agent_testing'
+  }
+  if ENV['DD_PIPELINE_ID']
+    vm_tags['pipeline_id'] = ENV['DD_PIPELINE_ID']
+  else
+    vm_tags['pipeline_id'] = 'nopipeline'
+  end
+%>
+
+provisioner:
+  name: chef_solo
+  product_name: chef
+  product_version: 14.12.9
+  install_strategy: always
+
+driver:
+  name: azurerm
+
+driver_config:
+  subscription_id: <%= ENV['AZURE_SUBSCRIPTION_ID'] %>
+  location: <%= location %>
+  <% if ENV['DD_PIPELINE_ID'] %>
+  azure_resource_group_suffix: pl<%= ENV['DD_PIPELINE_ID'] %>
+  <% else %>
+  azure_resource_group_suffix: plnone
+  <% end %>
+
+platforms:
+# Loop through two lists and output a total matrix of all possible platform + chef versions,
+# for both drivers
+
+
+<%
+    # TEST_PLATFORMS syntax is `short_name1,azure_full_qualified_name1|short_name2,azure_full_qualified_name1`
+    # TEST_PLATFORMS syntax is `short_name1,parent vhd folder,parent_vhd_name|...`
+    azure_test_platforms = ENV['TEST_PLATFORMS'].split('|').map { |p| p.split(',') }
+
+    sizes = [
+      "Standard_D1_v2",
+      "Standard_A1_v2",
+    ]
+
+    windows_sizes = [
+      "Standard_D2_v2"
+    ]
+
+    windows_platforms = []
+    sles11_platforms = []
+    sles15_platforms = []
+
+    idx = 0
+    azure_test_platforms.each do |platform|
```

Comment:
It took me a while to realize this is a loop, the code inside should be indented.